SOD.Common/Helpers/InputDetection.cs

```diff
@@ -1,19 +1,471 @@
+        {
+            if (suppressionEntryIds.Count == 0)
+                OnAxisStateChanged?.Invoke(this, new AxisInputDetectionEventArgs(actionName, interactionKey, axisValue));
+            else
+                OnSuppressedAxisStateChanged?.Invoke(this, new SuppressedAxisInputDetectionEventArgs(actionName, interactionKey, axisValue, suppressionEntryIds));
+        }
+
+        private List<ActionElementMap> keyboardAndMouseMappings;
+
+        /// <summary>
+        /// Contains all keyboard and mouse Rewired ActionElementMap mappings.
+        /// </summary>
+        public List<ActionElementMap> KeyboardAndMouseMappings
+        {
+            get
+            {
+                if (keyboardAndMouseMappings != null)
+                {
+                    return keyboardAndMouseMappings;
+                }
+                keyboardAndMouseMappings = new List<ActionElementMap>();
+                var controllers = new List<Rewired.Controller>();
+                Rewired.Controller keyboardController = RewiredPlayer.controllers.GetController(ControllerType.Keyboard, KEYBOARD_AND_MOUSE_CONTROLLER_ID);
+                controllers.Add(keyboardController);
+                Rewired.Controller mouseController = RewiredPlayer.controllers.GetController(ControllerType.Mouse, KEYBOARD_AND_MOUSE_CONTROLLER_ID);
+                controllers.Add(mouseController);
+                foreach (var controller in controllers)
+                {
+                    foreach (string category in CONTROLLER_MAP_CATEGORIES)
+                    {
+                        ControllerMap map = RewiredPlayer.controllers.maps.GetMap(controller.type, controller.id, category, "Default");
+                        keyboardAndMouseMappings.AddRange(map.AllMaps.ToList());
+                    }
+                }
+                return keyboardAndMouseMappings;
+            }
+        }
+
+        internal Dictionary<string, InputSuppressionEntry> InputSuppressionDictionary { get; set; } = new();
+
+        /// <summary>
+        /// Suppress vanilla game responses to an input, with an optional duration. Adds an input suppression entry.
+        /// <br>The input suppression entry will suppress vanilla game responses to the input while it or any entry tied to that input is present.</br>
+        /// <br>At the end of the duration (if provided), the input suppression entry will be automatically removed.</br>
+        /// </summary>
+        /// <param name="id">The id associated with this entry.</param>
+        /// <param name="keyCode">The KeyCode to be suppressed.</param>
+        /// <param name="duration">The duration that the input suppression lasts. (does not progress while the game is paused).</param>
+        /// <param name="overwrite">Whether to overwrite any existing entry with this id, applying the new duration.</param>
+        public void SetInputSuppressed(string id, KeyCode keyCode, TimeSpan? duration = null, bool overwrite = false)
```

Comment:
I think it makes more sense from a user point of view if we out an int id ourself?
Like we do with most builders, so the user doesn't have to come up with an id

Reply:
After some thought, I think I agree. I doesn't add anything to have it be user-specified since there is a maximum of one suppression entry per id. I'll change it